### ISSUE TITLE
Poor Color Contrast is Difficult to See

### ISSUE THREAD

**Issue by flywire** [ggio/gnucash_dark_everforest#1]
Poor color contrast make information difficult to see and comprehend. While visually impaired people are particularly affected by this issue good contrast improves resources for everyone.

Please improve contrast ratio.

Resources:
1. https://webaim.org/articles/contrast/
2. https://webaim.org/resources/contrastchecker/

Example: 

```css
/* Elementary colors */

@define-color yellow	    #b58900;
@define-color red	    #e04c56;
@define-color dark	    #2f383e;
@define-color dark2  	    #374247;
@define-color dark3  	    #22282c;
@define-color dark4	    #404c51;
@define-color fg-dark       #d3c6aa;
@define-color fg-hl	    #cdae6a;

/* Application wide colors */

@define-color	bg_color	@dark2;
@define-color	bg_highlight	@dark4;
@define-color	bg_dark		@dark;
@define-color	bg_darker	@dark3;
@define-color	fg_color	@fg-dark;
@define-color	fg_highlight	@yellow;
@define-color	fg_pop		@yellow;
@define-color	negative	@red;
```

The IDE uses default colors which mainly fail as shown using https://webaim.org/resources/contrastchecker/?fcolor=000000&bcolor=FFFFFF&api (replace fcolor and bcolor as required). The required ratio varies but starts at 4.5:

https://webaim.org/resources/contrastchecker/?fcolor=b58900&bcolor=2f383e&api
* yellow b58900 on  dark 2f383e
{"ratio":"3.72","AA":"fail","AALarge":"pass","AAA":"fail","AAALarge":"fail"}

**Comment by grgmiranda** [comment]
Yes indeed, thanks for your input.

I guess it's a common problem with these pastel tones color themes. Feels like a constant fight between making the contrast better without losing the "everforest" signature.

I've put some work on this: made the background darker and foreground brighter, had also to change some borders and other choices, as well as some refactoring. Please, let me know what do you think.

Most of bg/fg pairs now pass _most_ of the checker test (unless "WCAG AAA", wich was basically impossible for red and yellow colors without making they almost white).

I guess it's quite better now, but visually impaired people probably will have to chose some other theme or make their own adjustments.

**Comment by grgmiranda** [comment]
Closed by accident. Still open for input.

**Comment by flywire** [comment]
> visually impaired people probably will have to chose some other theme

Perhaps you misunderstand, this issue is just exaggerated for the visually impaired (the blind don't see at all):

> > Contrast and color use are vital to accessibility. Users... must be able to perceive content on the page.

The aim is to minimise the perception task for the brain so it can maximise understanding content. I am viewing this page in _Dark Default_ and it is very readable. Interface colours are not as easy as people imagine with change to one part affecting another part. For example, 665c1f4 sc_01.png change of red is good but the footer background is worse.

I aim only to raise awareness of the issue, https://webaim.org/articles/contrast/ demonstrates issues that commonly occur on the web. The contrast tool can help. While testing consider software used on many devices and resizable windows.

**Comment by flywire** [comment]
https://www.accessibility.com/glossary/advanced-perceptual-contrast-algorithm
> The WCAG 2.x contrast methods are being replaced for WCAG 3.0 APCA.

https://www.achecks.org/apca-accessible-colour-contrast-checker/

**Comment by grgmiranda** [comment]
Makes sense. Appreciate your feedback and references. Will put some thought and work on this.
